### ISSUE TITLE
Curse blocking requests: "Failed to find addon version number ..."

### ISSUE THREAD

**Issue by SveSop** [grrttedwards/wow-addon-updater#69]
Just started using this fork vs. the original dead one (that worked fine until today).
I figured since the updater did not work, i would pop over to a active project, but it seems either i have done something wrong, or curseforge changed something (again).

`Failed to find addon version number for game version: GameVersion.retail, https://www.curseforge.com/wow/addons/bartender4`
And so on and so forth..

```
Name                             Prev. Version                    New Version                      
────                             ─────────────                    ───────────                      
bartender4                       -----                            Unavailable                      
```
As i said, i might have missed some config step, but all my addons set in addons.txt are listed, and the url's seems to check out... just not the version.

**Comment by grrttedwards** [comment]
Yeah, Curse has been doing this for a few days now. Their CDN Cloudflare has been aggressively interrogating the API calls, because they're trying to determine whether or not you are coming from a browser. #68 is likely the same issue.

If you try again some time soon, it should work. It's not great.

In the meantime, I need to enhance the error reporting to at least differentiate between potential failures like this...

**Comment by grrttedwards** [comment]
#70 makes it more clear via logging, but a workaround is not known. What I have seen for busting Cloudflare requires your machine to have nodejs installed, and I'm not sure I like that prospect.

**Comment by vrooze** [comment]
Just to add into this a specific classic addon causing me grief.
https://www.curseforge.com/wow/addons/fishing-buddy-classic/files I've looked at the regex and the pattern doesn't match any of the classes. It's the only one in my list doing it :(

Also I hacked in some code to get cfscrape working locally  until you come up with a better solution.

**Comment by guppy42** [comment]
> Just to add into this a specific classic addon causing me grief.
> https://www.curseforge.com/wow/addons/fishing-buddy-classic/files I've looked at the regex and the pattern doesn't match any of the classes. It's the only one in my list doing it :(
> 
> Also I hacked in some code to get cfscrape working locally until you come up with a better solution.

The updater doesn't support beta releases unfortunately

**Comment by grrttedwards** [comment]
Hi @vrooze

Yes, the reason it won’t work for that specific addon is because it only
has a Beta version tag. Once the author publishes a Release version, then
the script can do its thing.

Of course, that’s provided that Curse ever allows bots to scrape the site
again!

Sorry about that.

**Comment by ritchiedalto** [comment]
This tool, I believe, can work by pointing your config urls to Curse or to the github of the addons. Has anyone tried the latter?

**Comment by grrttedwards** [comment]
Just as a PSA in the meantime: for anyone who is desperate to get some addons updated -- try to see if the addon is hosted on WoW Interface.

**Comment by mishugashu** [comment]
>  better yet, get it straight from the source control like GitHub.

Speaking of... is there a way to _not_ put the contents of github in a top level folder? Things like DBM have multiple addons that need to go in the top level AddOns directory. https://github.com/DeadlyBossMods/DeadlyBossMods

**Comment by Mayo77** [comment]
So what does this mean to a person that doesn't understand all this python lingo ? 
Is this error actually able to get fixed or no? 
Do i have to use wowmatrix now?  (puke)

**Comment by SveSop** [comment]
Right... so Curse and Wowace is basically the same thing, and not working atm.
That leaves wowinterface that misses/got old versions of a couple of my addons currently.. and github that do not have 3 of the addons i am using.

Meh.. That sux :)

**Comment by grrttedwards** [comment]
@mishugashu The "subfolder" functionality can be abused for this example. So -- while it's not pretty -- you can list each of the subfolders as an addon to update in your input file, and use the subfolder syntax to extract each...

e.g.

```
https://github.com/DeadlyBossMods/DeadlyBossMods|DBM-Azeroth-BfA
https://github.com/DeadlyBossMods/DeadlyBossMods|DBM-Brawlers
https://github.com/DeadlyBossMods/DeadlyBossMods|DBM-Core
https://github.com/DeadlyBossMods/DeadlyBossMods|DBM-CrucibleofStorms
https://github.com/DeadlyBossMods/DeadlyBossMods|DBM-DMF
https://github.com/DeadlyBossMods/DeadlyBossMods|DBM-DefaultSkin
https://github.com/DeadlyBossMods/DeadlyBossMods|DBM-EternalPalace
https://github.com/DeadlyBossMods/DeadlyBossMods|DBM-GUI
https://github.com/DeadlyBossMods/DeadlyBossMods|DBM-Party-BfA
https://github.com/DeadlyBossMods/DeadlyBossMods|DBM-StatusBarTimers
https://github.com/DeadlyBossMods/DeadlyBossMods|DBM-Uldir
https://github.com/DeadlyBossMods/DeadlyBossMods|DBM-WorldEvents
https://github.com/DeadlyBossMods/DeadlyBossMods|DBM-ZuldazarRaid
```

If that ends up being a real use case, maybe it should be polished up a bit. But like you said, that might be a waste of time because who knows what structure the authors will use on GitHub.

**Comment by grrttedwards** [comment]
If everyone wouldn't mind taking a look at https://github.com/grrttedwards/wow-addon-updater/pull/73

This brings in `cfscrape`, which busts the Cloudflare protection (nice job, Curse... it was _really hard_ to circumvent your scripts :eyeroll: )

Note that this REQUIRES Node installed. Please try it out and give me some feedback so I can get this in.

**Comment by joshrsmith** [comment]
@grrttedwards #73 is working for me! Previously had same problems as described in this thread.

All I had to do was rerun `pipenv`. I already had `node` installed.

**Comment by grrttedwards** [comment]
Merged #73 and released [v1.2.0](https://github.com/grrttedwards/wow-addon-updater/releases/tag/v1.2.0)
